src/style/palette/index.js

```diff
@@ -0,0 +1,28 @@
+import tint from '../utils/tint';
+import shade from '../utils/shade';
+
+const cachedFunc = cb => (cache = {}) => (weight) => {
+  if (cache[weight]) { return cache[weight]; }
+  const color = cb(weight);
+  cache[weight] = color;
+  return color;
+};
+
+const palette = (config) => {
+  const baseNames = Object.keys(config);
+
+  const funcs = baseNames.reduce((acc, baseName) => {
+    const tintBy = tint(config[baseName]),
+        shadeBy = shade(config[baseName]);
+
+    acc[`${baseName}Tint`] = cachedFunc(tintBy)();
+    acc[`${baseName}Shade`] = cachedFunc(shadeBy)();
+
+    return acc;
+  }, {});
+
+  return { ...funcs, ...config };
```

Comment:
are that config entries needed in a palette object? Config could be imported directly from ```color-config```.

Reply:
The palette needs to pass the base colours along with the functions, it seemed tidier to draw them into the palette here, otherwise they will have to be included into each individual theme file.

Reply:
having both methods and entries in one object could be confusing, when their names are nouns.
I think that renaming these methods to verbs would be more meaningful.

Reply:
`tint` and `shade` are verbs, so to me the naming works. Agreed that it'd probably better to separate out the base colours, so I'll do that.